models/maat/R/met2model.MAAT.R

```diff
@@ -94,54 +95,78 @@ met2model.MAAT <- function(in.path, in.prefix, outfolder, start_date, end_date,
       frac.day <- nc$dim$time$vals
       sec <- udunits2::ud.convert(sec, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
 
-      dt <- PEcAn.utils::seconds_in_year(year) / length(sec)
+      dt <- PEcAn.utils::seconds_in_year(year, leap_year) / length(sec)
 
-      tstep <- round(86400 / dt)
-      dt    <- 86400 / tstep
+      tstep <- round(day_secs / dt)
+      dt    <- day_secs / tstep
 
-      ## extract required MAAT driver variables names(nc$var)
+      ### extract required MAAT driver variables names(nc$var)
       lat  <- ncdf4::ncvar_get(nc, "latitude")
       lon  <- ncdf4::ncvar_get(nc, "longitude")
+      
+      # Air temperature
       Tair <- ncdf4::ncvar_get(nc, "air_temperature")  ## in Kelvin
+      Tair_C <- udunits2::ud.convert(Tair, "K", "degC") ## in degC
+      
+      # Precipitation
       Rain <- ncdf4::ncvar_get(nc, "precipitation_flux")  ## 'kg/m^2/s'
+      
+      # Get atmospheric pressure
+      Atm_press <- ncdf4::ncvar_get(nc,"air_pressure") ## in Pa
 
-      # get humidity vars (NOTE:later add VPD here!!)
+      # get humidity vars
       RH_perc <- ncdf4::ncvar_get(nc, "relative_humidity")  ## RH Percentage
-
+      Qair <- try(ncdf4::ncvar_get(nc, "specific_humidity"), silent = TRUE)  #humidity (kg/kg)
+      SVP <- udunits2::ud.convert(PEcAn.data.atmosphere::get.es(Tair_C), "millibar", "Pa")  ## Saturation vapor pressure
+      VPD <- try(ncdf4::ncvar_get(nc, "water_vapor_saturation_deficit"), silent = TRUE)  ## in Pa
+      if (!is.numeric(VPD)) {
+        VPD <- SVP * (1 - PEcAn.data.atmosphere::qair2rh(Qair, Tair_C))
```

Comment:
not required, but it seems like a more reusable solution would be to update `get.vpd` to take Tair and either RH or SH